src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/ASEReadCounter.java

```diff
@@ -0,0 +1,286 @@
+        defaultFilters.add(ReadFilterLibrary.VALID_ALIGNMENT_END);
+        defaultFilters.add(ReadFilterLibrary.HAS_READ_GROUP);
+        defaultFilters.add(ReadFilterLibrary.HAS_MATCHING_BASES_AND_QUALS);
+        defaultFilters.add(ReadFilterLibrary.SEQ_IS_STORED);
+        defaultFilters.add(ReadFilterLibrary.NOT_DUPLICATE);
+        defaultFilters.add(ReadFilterLibrary.NOT_SECONDARY_ALIGNMENT);
+        defaultFilters.add(new ReadFilterLibrary.MappedReadFilter());
+        return defaultFilters;
+    }
+
+    @Override
+    public void onTraversalStart() {
+        try {
+            outputStream = OUTPUT_FILE != null ? new PrintStream(OUTPUT_FILE) : System.out;
+        } catch (FileNotFoundException e) {
+            throw new UserException.CouldNotReadInputFile(OUTPUT_FILE, e);
+        }
+
+        // Check the output format
+        boolean goodOutputFormat = false;
+        for (final OUTPUT_FORMAT f : OUTPUT_FORMAT.values()) {
+            goodOutputFormat = goodOutputFormat || f.equals(outputFormat);
+        }
+
+        if (!goodOutputFormat) {
+            throw new IllegalArgumentException("Improper output format. Can be one of TABLE, RTABLE, CSV. Was " + outputFormat);
+        }
+
+        if (outputFormat.equals(OUTPUT_FORMAT.CSV)) {
+            separator = ",";
+        }
+        final String header = "contig" + separator + "position" + separator + "variantID" + separator + "refAllele" + separator + "altAllele" + separator + "refCount" + separator + "altCount" + separator + "totalCount" + separator + "lowMAPQDepth" + separator + "lowBaseQDepth" + separator + "rawDepth" + separator + "otherBases" + separator + "improperPairs";
+        outputStream.println(header);
+    }
+
+    @Override
+    public void apply(AlignmentContext alignmentContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        final String contig = alignmentContext.getContig();
+        final long position = alignmentContext.getPosition();
+
+        final char refAllele = (char) referenceContext.getBase();
+
+        final List<VariantContext> VCs = featureContext.getValues(variants);
+        if (VCs != null && VCs.size() > 1)
+            throw new UserException("More then one variant context at position: " + contig + ":" + position);
+        if (VCs == null || VCs.isEmpty())
+            return;
+
+        final VariantContext vc = VCs.get(0);
+        if (!vc.isBiallelic()) {
+            logger.warn("Ignoring site: cannot run ASE on non-biallelic sites: " + vc.toString());
+            return;
+        }
+
+        if (vc.getHetCount() < 1) {
+            logger.warn("Ignoring site: variant is not het at postion: " + contig + ":" + position);
+            return;
+        }
+
+        if (vc.getNAlleles() == 1 || vc.getAlternateAllele(0).getBases().length == 0)
+            throw new UserException("The file of variant sites must contain heterozygous sites and cannot be a GVCF file containing <NON_REF> alleles.");
```

Comment:
This case should have a test too.  You might want ones for the warning cases that just check that they don't crash, but I think those can slide if you don't want to bother.